tcp-native/src/main/java/org/ballerinalang/stdlib/tcp/TcpListenerHandler.java

```diff
@@ -54,4 +57,22 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
         Dispatcher.invokeOnError(tcpService, cause.getMessage());
     }
+
+    @Override
+    public void channelWritabilityChanged(ChannelHandlerContext ctx) throws Exception {
+        if (ctx.channel().isWritable() && writeFlowControllers.size() > 0) {
+            WriteFlowController writeFlowController = writeFlowControllers.getFirst();
```

Comment:
I think we need a while loop here.

Reply:
+1

Reply:
added while (writeFlowControllers.size() > 0)